src/SDK/SmartDetectors.Extensions/Extensions/AlertExtensions.cs

```diff
@@ -100,7 +91,7 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
 
             string id = string.Join("##", alert.GetType().FullName, JsonConvert.SerializeObject(request), JsonConvert.SerializeObject(alert)).ToSha256Hash();
             string resourceId = alert.ResourceIdentifier.ToResourceId();
-            string correlationHash = string.Join("##", predicates.OrderBy(x => x.Key).Select(x => x.Key + "|" + x.Value)).ToSha256Hash();
+            string correlationHash = string.Join("##", predicates.OrderBy(x => x.Key).Select(x => x.Key + "|" + x.Value.ToString())).ToSha256Hash();
```

Comment:
>ToString() [](start = 120, length = 10)

InvariantCulture?

Reply:
object.ToString does not accept any parameters

---
In reply to: [234989473](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r234989473) [](ancestors = 234989473)

Reply:
Consider adding a method that will check the type and add culture when applicable 

---
In reply to: [235285958](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/27#discussion_r235285958) [](ancestors = 235285958,234989473)